store.go

```diff
@@ -71,6 +71,20 @@ type metadataStore interface {
 	rwMetadataStore
 }
 
+// PutLayerFromStagingDirectoryOptions contains options to pass to PutLayerFromStagingDirectory
+type PutLayerFromStagingDirectoryOptions struct {
+	ID           string
+	ParentLayer  string
+	Names        []string
+	MountLabel   string
+	Writeable    bool
+	LayerOptions *LayerOptions
+
+	StagingDirectory string
```

Comment:
Would `StagingDirectory` ever differ from `DiffOutput.Target`? If not, can this be handled internally without callers having to care (and without callers possibly breaking this)?

OTOH `CleanupStagingDirectory` forces c/image to know about `.Target` anyway…